system/Entity.php

```diff
@@ -207,7 +248,19 @@ public function toRawArray(bool $onlyChanged = false): array
 				continue;
 			}
 
-			$return[$key] = $this->attributes[$key];
+			if ($deep)
+			{
+				if ($value instanceof Entity)
+				{
+					$value = $value->toRawArray($onlyChanged, $deep);
+				}
+				else if (is_callable([$value, 'toRawArray']))
```

Comment:
You missed one spot :)
```suggestion
				elseif (is_callable([$value, 'toRawArray']))
```